src/UI/state/speciesInformation/speciesInformationSlice.ts

```diff
@@ -1,14 +1,55 @@
 import { createSlice } from '@reduxjs/toolkit';
 import { SpeciesInformation } from '../state.types';
+import { getAllSpecies } from './actions/getAllSpecies';
 
 export interface SpeciesInformationState {
   currentInfoForEditing: SpeciesInformation | null;
   loading: boolean;
+  currentInfoDetails: SpeciesInformation | null;
+  speciesDict: {
+    items: SpeciesItems[];
+    total: number;
+  };
+  speciesInfoStatus: string;
+  speciesListOptions: {
+    page: number;
+    rowsPerPage: number;
+    orderBy: string;
+    order: 'asc' | 'desc';
+    startId: number | null;
+    endId: number | null;
+    textFilter: string;
+  };
+}
+
+export interface SpeciesItems {
```

Comment:
This is very similar to `SpeciesInformation` - can you extend that interface here?